app/src/main/java/com/arctouch/codechallenge/view/home/MovieDetailActivity.java

```diff
@@ -0,0 +1,61 @@
+package com.arctouch.codechallenge.view.home;
+
+import android.os.Bundle;
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AppCompatActivity;
+import android.text.TextUtils;
+
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import com.arctouch.codechallenge.R;
+import com.arctouch.codechallenge.data.model.Movie;
+import com.arctouch.codechallenge.util.MovieImageUrlBuilder;
+import com.bumptech.glide.Glide;
+import com.bumptech.glide.request.RequestOptions;
+
+public class MovieDetailActivity extends AppCompatActivity {
+
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        setContentView(R.layout.movie_detail_activity_layout);
+
+        ImageView backDropImage = findViewById(R.id.detail_backdrop_image);
+        ImageView posterImage = findViewById(R.id.detail_poster_image);
+
+        TextView movieTitle = findViewById(R.id.detail_movie_name);
+        TextView movieGenres = findViewById(R.id.detail_movie_genres);
+        TextView movieReleaseDate = findViewById(R.id.detail_release_date);
+        TextView movieOverview = findViewById(R.id.detail_overview);
+
+
+        if (getIntent().getExtras().containsKey("movie")) {
```

Comment:
AndroidStudio warns a `NullPointerException` could be thrown here. What would be your strategy for handling (avoiding, treating...) that?

Reply:
I would check null the intent and the extras before trying to use them.